### PR TITLE
expose the bank vault to the script engine

### PR DESCRIPTION
**Pull request by mekolat**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [X] I have followed [proper Hercules code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
* Expose `map_session_data->status.bank_vault` as `BankVault` to the script engine
* Expose `MAX_BANK_ZENY` to the script engine
* Update the documentation to reflect those changes

<br>

### Why expose the vault?
* Allows NPCs to manipulate the vault, so players with a client that does not support the vault packets can still deposit and withdraw money through a NPC.

<br>

### Example
```c
if (Zeny >= .@val && (BankVault + .@val) <= MAX_BANK_ZENY) {
    Zeny -= .@val;
    BankVault += .@val;
}
```

<br><br>

**Affected Branches:** `master`
**Type:** [enhancement](https://github.com/HerculesWS/Hercules/labels/type%3Aenhancement)
**Compopents:** [core](https://github.com/HerculesWS/Hercules/labels/component%3Acore), [script engine](https://github.com/HerculesWS/Hercules/labels/component%3Acore%3Ascriptengine), [documentation](https://github.com/HerculesWS/Hercules/labels/component%3Adocumentation)
**Tested on:** [ManaPlus](https://manaplus.org)
Needs testing on the official client

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

---
<sub>linked to https://gitlab.com/evol/serverdata/merge_requests/106</sub>

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1717)

File: src/map/map.h [hunk 1]
```diff
@@ -553,6 +553,7 @@ enum status_point_types { //we better clean up this enum and change it name [Hem
 	SP_MOD_EXP=125,
 	SP_MOD_DROP=126,
 	SP_MOD_DEATH=127,
+	SP_BANKVAULT=128,
```
Comment (4144) [inline src/map/map.h]: not sure is possible add any custom contants here. I think they related to aegis or client constants?

Comment (dastgir) [inline src/map/map.h]: The first 100 are used by client, all constant above 100 are not related to client/aegis.

**Comment by mekolat** [comment]
@MishimaHaruna  @4144  is there anything you want me to change?

**Comment by MishimaHaruna** [comment]
Nothing to change, looks good to me.

I'm merging it, thank you!

**Pull request merged**